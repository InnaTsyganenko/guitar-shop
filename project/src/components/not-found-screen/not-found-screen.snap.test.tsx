import { render } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import HistoryRouter from '../history-route/history-route';
import { Provider } from 'react-redux';
import { configureMockStore } from '@jedmao/redux-mock-store';
import thunk from 'redux-thunk';
import NotFoundScreen from './not-found-screen';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

const store = mockStore({
  DATA: {isDataLoaded: true},
  GUITARS: {pickedId: 1, currentPageCatalog: 1, isModalOpen: true},
});

describe('Component: NotFoundScreen', () => {
  it('should render correctly', () => {
    const history = createMemoryHistory();

    const {container} = render(
      <Provider store={store}>
        <HistoryRouter history={history}>
          <NotFoundScreen />
        </HistoryRouter>
      </Provider>,
    );

    expect(container).toMatchSnapshot();
  });
});
