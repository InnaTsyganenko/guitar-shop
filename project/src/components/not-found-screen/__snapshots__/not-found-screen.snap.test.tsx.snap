// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: NotFoundScreen should render correctly 1`] = `
<div>
  <div
    class="wrapper"
    data-testid="wrapper"
  >
    <main
      class="page-content"
    >
      <div
        class="container"
      >
        <p>
          Oooops!
        </p>
        <h2>
          404. Page not found.
        </h2>
        <h3>
          <a
            href="/"
            style="text-decoration: underline;"
          >
            Вернуться на главную
          </a>
          ?
        </h3>
      </div>
    </main>
  </div>
</div>
`;
